tests/e2e_test.go

```diff
@@ -932,3 +932,74 @@ func TestKeyPOP(t *testing.T) {
 		t.Fatal("verification should fail")
 	}
 }
+
+func TestRotateRootKeyTwiceAfter(t *testing.T) {
+	// This tests a corner case of key rotation. If a rotation occurs on
+	// v2, then we expect two signature placeholders for staging the v2 root.
+	// For the next v3, if there are no other root changes we should only
+	// expect one. Under the hood, this is testing that InitCmd always stages
+	// a new root before calculating placeholer signatures.
+	ctx := context.Background()
+	stack := newRepoTestStack(ctx, t)
+	stack.addTarget(t, "foo.txt", "abc", nil)
+	rootKeyRef1 := stack.genKey(t, true)
+	rootKeyRef2 := stack.genKey(t, true)
+
+	// Initialize succeeds
+	if err := app.InitCmd(ctx, stack.repoDir, 1,
+		stack.targetsConfig, stack.repoDir, stack.snapshotRef, stack.timestampRef); err != nil {
+		t.Fatal(err)
+	}
+
+	// Sign root & targets with key 1
+	rootSigner1 := stack.getSigner(t, rootKeyRef1)
+	if err := app.SignCmd(ctx, stack.repoDir, []string{"root", "targets"},
+		rootSigner1, false); err != nil {
+		t.Fatal(err)
+	}
+	pubKey1, err := keys.ConstructTufKey(ctx, rootSigner1)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Sign snapshot and timestamp
+	stack.snapshot(t)
+	stack.timestamp(t)
+	stack.publish(t)
+
+	// Now remove the second key.
+	stack.removeHsmKey(t, rootKeyRef2)
+	// Create a new root.
+	if err := app.InitCmd(ctx, stack.repoDir, 1,
```

Comment:
Would it make sense to verify here (and the above init) that the expected number of signatures are the correct too as it's done below?

Reply:
Yes :) It was covered by an earlier test but I was actually just being lazy here.